### PR TITLE
FSE: fix Template Part name in Block Navigator

### LINKED ISSUE #36508: FSE: template part generic name appears in block inspector
When you open block inspector on an FSE page you can see the following:

image

As @shaunandrews suggested, a better solution here would be to show the label instead of the block name.

Let's follow this up with a core PR (and issue possibly) after the launch.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
#### Changes proposed in this Pull Request

* added __experimentalDisplayName and label attribute for header and footer

Before - 
Screen Shot 2019-10-28 at 6 34 54 PM

After - 
Screen Shot 2019-10-28 at 6 34 10 PM

#### Testing instructions

Testing with local Gutenberg environment with latest Master branch.
1. Run this PR
2. edit a Page
3. verify that "Header" and "Footer" are titles in the Block Navigation component.

Fixes #36508

### CODE REVIEW

**Comment by noahtallen** [comment]
Note for reviewers that this experimental feature might not be on the current dotcom Gutenberg version yet. AFAIK this shouldn't be a problem and the code shouldn't crash, but that's worth a check.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template.php [hunk 1]
```diff
@@ -161,9 +161,9 @@ public function get_page_template_content() {
 			return null;
 		}
 
-		return "<!-- wp:a8c/template {\"templateId\":$header_id,\"className\":\"site-header site-branding\"} /-->" .
+		return "<!-- wp:a8c/template {\"templateId\":$header_id,\"label\":\"Header\",\"className\":\"site-header site-branding\"} /-->" .
```
Comment (Copons) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template.php]: Here (and later for the footer) we are hardcoding the template part names, while instead we should translate them.

For example:
```php
return '<!-- wp:a8c/template {"label":"' . __( 'Header', 'full-site-editing' ) . '" } /-->';
```

---

In the future, when we start adding proper localized support (or having more template parts with unpredictable names), we might end up replacing the simple translation function with a different approach.
For example, the title of the template part post returned by the API could be already translated, so we could do something lie this:
```php
return '<!-- wp:a8c/template {"label":"' . get_the_title( $header_id ) . '" } /-->';
```

Just to be clear: **this wouldn't work right now**, it's just an idea of what could eventually be.
But now that we have hardcoded template parts, it's plenty enough to just simply translate them.

**Comment by Copons** [comment]
This works as expected! 🎉 
I've requested a minor change that though we should tackle before merging.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template.php (hunk 1 above)
Comment (Addison-Stavlo) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template.php]: That is great advice, I just added it in the commit as seen below.
Thanks!

**Pull request merged**